From 49d7d01e6cfcd86cdb8d60ec7a45c2aab39b4590 Mon Sep 17 00:00:00 2001
From: Tiejun Chen <tiejun.chen@windriver.com>
Date: Thu, 8 Jul 2010 16:28:29 +0800
Subject: [PATCH] fsl_p4080ds: Fix incorrect DMA memory size when unmap

We should use the same DMA memory size while map/unmap DMA entity.

After refer to the previous code:
---
        outbusaddr = dma_map_single(dev, jqp->outring,
                                    sizeof(struct jq_outentry) * JOBQ_DEPTH,
...

So update size with "sizeof(struct jq_outentry) * JOBQ_DEPTH" when unmap that.

Signed-off-by: Tiejun Chen <tiejun.chen@windriver.com>
---
 drivers/crypto/caam/jq.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/drivers/crypto/caam/jq.c b/drivers/crypto/caam/jq.c
index 0e287b5..0b20f53 100644
--- a/drivers/crypto/caam/jq.c
+++ b/drivers/crypto/caam/jq.c
@@ -386,7 +386,7 @@ int caam_jq_init(struct device *dev)
 					 sizeof(u32 *) * JOBQ_DEPTH,
 					 DMA_BIDIRECTIONAL);
 			dma_unmap_single(dev, outbusaddr,
-					 sizeof(u32 *) * JOBQ_DEPTH,
+					 sizeof(struct jq_outentry) * JOBQ_DEPTH,
 					 DMA_BIDIRECTIONAL);
 			kfree(jqp->inpring);
 			kfree(jqp->outring);
-- 
1.6.5.2

